onnxruntime/core/providers/dml/DmlExecutionProvider/src/External/DirectMLHelpers/StringifyHelpers.h

```diff
@@ -0,0 +1,54 @@
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+#pragma once
+
+#pragma warning(push)
+#pragma warning(disable:4499)
+
+// This should only include ToString functions for enums which are not part of DirectMLSchema.json
+namespace StringifyHelpers
+{
+    template <typename T>
+    static const char* ToString(T value)
+    {
+#ifndef WAI_BUILD_LINUX
+        // Clang will instantiate this template even if it isn't used,
+        // so this static_assert will always fire and break the build.
+        static_assert(false, "Not implemented for this type");
+#endif
+    }
+
+    template<>
+    static const char* ToString(DmlStage value)
+    {
+        switch (value)
+        {
+        case DmlStage::Input:      return "input";
+        case DmlStage::Output:     return "output";
+        case DmlStage::Persistent: return "the persistent resource";
+        case DmlStage::GlobalMemoryTemporary:  return "the global memory temporary resource";
+        case DmlStage::FixedFunctionCacheMemoryTemporary:  return "the fixed function cache memory temporary resource";
+        case DmlStage::DspCacheMemoryTemporary:  return "the DSP cache memory temporary resource";
+
+        default:
+            return "<unknown>";
```

Comment:
[confetti nit] If you're indenting some of them, then indent all or none.

```
        case DmlStage::Input:                               return "input";
        case DmlStage::Output:                              return "output";
        case DmlStage::Persistent:                          return "the persistent resource";
        case DmlStage::GlobalMemoryTemporary:               return "the global memory temporary resource";
        case DmlStage::FixedFunctionCacheMemoryTemporary:   return "the fixed function cache memory temporary resource";
        case DmlStage::DspCacheMemoryTemporary:             return "the DSP cache memory temporary resource";
        default:                                            return "<unknown>";
```

or

```
        case DmlStage::Input: return "input";
        case DmlStage::Output: return "output";
        case DmlStage::Persistent: return "the persistent resource";
        case DmlStage::GlobalMemoryTemporary: return "the global memory temporary resource";
        case DmlStage::FixedFunctionCacheMemoryTemporary: return "the fixed function cache memory temporary resource";
        case DmlStage::DspCacheMemoryTemporary: return "the DSP cache memory temporary resource";
        default: return "<unknown>";
```

Reply:
Thank you. This was copy/pasted from DML repo. I should have corrected in both places. I will update the DML repo as well.